src/androidTest/java/com/nutomic/syncthingandroid/test/syncthing/AppConfigReceiverTest.java

```diff
@@ -0,0 +1,107 @@
+package com.nutomic.syncthingandroid.test.syncthing;
+
+import android.content.Intent;
+import android.preference.PreferenceManager;
+import android.test.AndroidTestCase;
+
+import com.nutomic.syncthingandroid.syncthing.AppConfigReceiver;
+import com.nutomic.syncthingandroid.syncthing.SyncthingService;
+import com.nutomic.syncthingandroid.test.MockContext;
+
+/**
+ * Test the correct behaviour of the AppConfigReceiver
+ *
+ * Created by sqrt-1674 on 27.03.16.
+ */
+public class AppConfigReceiverTest extends AndroidTestCase {
+    private AppConfigReceiver mReceiver;
+    private MockContext mContext;
+
+    @Override
+    protected void setUp() throws Exception {
+        super.setUp();
+        mReceiver = new AppConfigReceiver();
+        mContext = new MockContext(getContext());
+    }
+
+    @Override
+    protected void tearDown() throws Exception {
+        PreferenceManager.getDefaultSharedPreferences(mContext).edit().clear().commit();
+        super.tearDown();
+    }
+
+
+    /**
+     * Test starting and stopping the Syncthing-Service if "always run in background" is enabled
+     * In this case starting the service is allowed
+     *
+     * Stopping the service is not allowed. In that case a notification is created.
+     * (This crashes in this test context which indicates that the test succeeded.)
+     */
+    public void testStartStopSyncthingServiceBackground() {
+        PreferenceManager.getDefaultSharedPreferences(mContext)
+                .edit()
+                .putBoolean(SyncthingService.PREF_ALWAYS_RUN_IN_BACKGROUND, true)
+                .commit();
+
+
+        Intent intent = new Intent(new Intent(mContext, AppConfigReceiver.class));
+        intent.putExtra(AppConfigReceiver.ACTION_SYNCTHINGSERVIVE, "start");
+
+        mReceiver.onReceive(mContext, intent);
+        assertEquals("Start SyncthingService Background", 1, mContext.getReceivedIntents().size());
+        assertEquals("Start SyncthingService Background", "com.nutomic.syncthingandroid.syncthing.SyncthingService", mContext.getReceivedIntents().get(0).getComponent().getClassName());
+
+        mContext.clearReceivedIntents();
+
+        intent = new Intent(new Intent(mContext, AppConfigReceiver.class));
+        intent.putExtra(AppConfigReceiver.ACTION_SYNCTHINGSERVIVE, "stop");
+
+            // onReceive now creates a notification.
+            // This is expected to throw an exception as creating a notification is not supported in this Test-Context
+            // so an exception is an indicator of success in this case.
+        try {
+            mReceiver.onReceive(mContext, intent);
+            throw new Exception("Stop SyncthingService Background - didn't fail to create the notification");
+        } catch (Exception e) {
```

Comment:
Can you catch the exact exception that is thrown? This would actually catch the exception you were throwing above, or also any assert.